openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -32,6 +38,111 @@ def get_identity(self, data):
+        """
+        Overwrite to_native to handle custom logic since we are serializing two models as one here
+        :param user: User object
+        :return: Dict serialized account
+        """
+        profile = user.profile
+
+        data = {
+            "username": user.username,
+            "url": self.context.get('request').build_absolute_uri(
+                reverse('accounts_api', kwargs={'username': user.username})
+            ),
+            "email": user.email,
+            "date_joined": user.date_joined,
+            "is_active": user.is_active,
+            "bio": AccountLegacyProfileSerializer.convert_empty_to_None(profile.bio),
+            "country": AccountLegacyProfileSerializer.convert_empty_to_None(profile.country.code),
+            "profile_image": self._get_profile_image(profile, user),
+            "time_zone": None,
+            "language_proficiencies": LanguageProficiencySerializer(
+                profile.language_proficiencies.all(),
+                many=True
+            ).data,
+            "name": profile.name,
+            "gender": AccountLegacyProfileSerializer.convert_empty_to_None(profile.gender),
+            "goals": profile.goals,
+            "year_of_birth": profile.year_of_birth,
+            "level_of_education": AccountLegacyProfileSerializer.convert_empty_to_None(profile.level_of_education),
+            "mailing_address": profile.mailing_address,
+            "requires_parental_consent": profile.requires_parental_consent(),
+        }
+
+        return self._filter_fields(
+            self._visible_fields(profile, user),
+            data
+        )
+
+    def _get_profile_image(self, user_profile, user):
+        """ Returns metadata about a user's profile image. """
+        data = {'has_image': user_profile.has_profile_image}
+        urls = get_profile_image_urls_for_user(user, self.context.get('request'))
+        data.update({
+            '{image_key_prefix}_{size}'.format(image_key_prefix=PROFILE_IMAGE_KEY_PREFIX, size=size_display_name): url
+            for size_display_name, url in urls.items()
+        })
+
+        return data
+
+    def _visible_fields(self, user_profile, user):
+        """
+        Return what fields should be visible based on user settings
+
+        :param user_profile: User profile object
+        :param user: User object
+        :return: whitelist List of fields to be shown
+        """
+
+        if self.custom_fields:
+            return self.custom_fields
+
+        profile_visibility = self._get_profile_visibility(user_profile, user)
+
+        if profile_visibility == ALL_USERS_VISIBILITY:
+            return self.configuration.get('shareable_fields')
+        else:
+            return self.configuration.get('public_fields')
+
+    def _get_profile_visibility(self, user_profile, user):
+        """Returns the visibility level for the specified user profile."""
+        if user_profile.requires_parental_consent():
```

Comment:
I don't have my work computer to easily look... But I feel like this logic (this whole method) must exist elsewhere. Is there any code that can be shared?